### PR TITLE
Add more compiler warnings

### PR DESCRIPTION
**Pull request by saraedum**
these are the flags we're using at libflatsurf.

### CODE REVIEW

File: poly_extra/bisection_step_arb.c [hunk 1]
```diff
@@ -16,6 +16,8 @@ void _fmpz_poly_bisection_step_arf(arf_t l, arf_t r, const fmpz * pol, slong len
     arb_t b, c;
     arf_t m;
 
+    (void)(sr); // unused parameter
```
Comment (saraedum) [inline poly_extra/bisection_step_arb.c:19]: @videlec: Is this intentional?

File: build.warn [hunk 2]
```diff
@@ -0,0 +1,508 @@
...
+                                                                     ^
+../e-antic/poly_extra/fmpz_poly_isolate_real_roots.c:200:69: warning: conversion to 'mp_limb_t {aka long unsigned int}' from 'mp_limb_signed_t {aka long int}' may change the sign of the result [-Wsign-conversion]
+                     fmpq_mul_2exp(exact_roots + i, exact_roots + i, k);
+                                                                     ^
+../e-antic/poly_extra/fmpz_poly_isolate_real_roots.c:202:69: warning: conversion to 'mp_limb_t {aka long unsigned int}' from 'mp_limb_signed_t {aka long int}' may change the sign of the result [-Wsign-conversion]
+                     fmpq_div_2exp(exact_roots + i, exact_roots + i, -k);
+                                                                     ^
+../e-antic/poly_extra/fmpz_poly_isolate_real_roots.c: In function '_fmpz_poly_isolate_real_roots_0_1_vca':
+../e-antic/poly_extra/fmpz_poly_isolate_real_roots.c:77:31: warning: this statement may fall through [-Wimplicit-fallthrough=]
+                 (*n_intervals)++;
```
Comment (saraedum) [inline build.warn:23]: @videlec: Is this intentional? (The comments do not seem to indicate it.)

**Comment by saraedum** [comment]
@videlec: I marked the warnings that I am not sure about. The others I can probably fix myself.

File: poly_extra/bisection_step_arb.c (hunk 1 above)
Comment (videlec) [reply poly_extra/bisection_step_arb.c:19]: `sl` (left sign) is indeed enough.

File: build.warn (hunk 2 above)
Comment (videlec) [reply build.warn:23]: I think this is intentional. The `case 0` corresponds to do backtrack in the dichotomy. Even if you find a root you want to perform this backtracking. Is it better to write it without fall through?

File: e-antic/e-antic.h.in [hunk 3]
```diff
@@ -22,13 +22,13 @@ extern "C" {
 
 /* e-antic version number */
 
-#define __E_ANTIC_VERSION 1
```
Comment (saraedum) [inline e-antic/e-antic.h.in:25]: These names are reserved: https://www.gnu.org/software/libc/manual/html_node/Reserved-Names.html

File: poly_extra/evaluate.c [hunk 4]
```diff
@@ -16,12 +16,10 @@
 #undef NUMBER
 
 #define NUMBER arf
-#define ADD(z,x,y,prec) arf_mul(z,x,y,prec,ARF_RND_NEAR)
```
Comment (saraedum) [inline poly_extra/evaluate.c:19]: Unused (and likely not what was intended.)

File: e-antic/renf_elem.h [hunk 5]
```diff
@@ -90,13 +90,13 @@ int renf_elem_cmp_si(renf_elem_t a, const slong b, renf_t nf)
...
 static __inline__
-int renf_elem_cmp_ui(renf_elem_t a, const slong b, renf_t nf)
+int renf_elem_cmp_ui(renf_elem_t a, const ulong b, renf_t nf)
 {
     int s;
     fmpq_t bq;
 
     fmpq_init(bq);
-    fmpz_set_si(fmpq_numref(bq), b);
+    fmpz_set_ui(fmpq_numref(bq), b);
```
Comment (saraedum) [inline e-antic/renf_elem.h:99]: That was probably wrong?

File: poly_extra/poly_evaluate_template.h [hunk 6]
```diff
@@ -15,7 +15,7 @@
 #define PASTE(X,Y) PASTE2(X,Y)
 
 #define NUM_T PASTE(NUMBER, _t)
-#define _FMPZ_POLY_EVALUATE PASTE(_fmpz_poly_evaluate_, NUMBER)
```
Comment (saraedum) [inline poly_extra/poly_evaluate_template.h:18]: These names are reserved.

File: poly_extra/sgn_at_half.c [hunk 7]
```diff
@@ -1,45 +0,0 @@
...
-    the terms of the GNU Lesser General Public License (LGPL) as published
-    by the Free Software Foundation; either version 2.1 of the License, or
-    (at your option) any later version.  See <http://www.gnu.org/licenses/>.
-*/
-
-#include <e-antic/poly_extra.h>
-
-/* the function below returns the sign of pol(1/2) */
-/* TODO: avoid allocation of x and y               */
-int _fmpz_poly_sgn_at_half(fmpz * pol, slong len)
```
Comment (saraedum) [inline poly_extra/sgn_at_half.c:23]: Not used anywhere.

File: renf/randtest.c [hunk 8]
```diff
@@ -55,7 +57,7 @@ void renf_randtest(renf_t nf, flint_rand_t state, slong len, slong prec, mp_bitc
 
     /* pick a random real root */
     c_array = _fmpz_vec_init(p->length);
-    k_array = malloc((p->length) * sizeof(slong));
```
Comment (saraedum) [inline renf/randtest.c:58]: Any reason we don't want to flint_malloc here?

File: renf_elem/cmp_fmpq.c [hunk 9]
```diff
@@ -12,33 +12,6 @@
 
 #include <e-antic/renf_elem.h>
 
-/* TODO: move it to ANTIC? */
-int _nf_equal_fmpq(const nf_elem_t a, const fmpq_t b, const nf_t nf)
```
Comment (saraedum) [inline renf_elem/cmp_fmpq.c:16]: not used anywhere.

File: renf_elem/sgn.c [hunk 10]
```diff
@@ -44,9 +44,5 @@ int renf_elem_sgn(renf_elem_t a, renf_t nf)
         renf_refine_embedding(nf, prec);
         renf_elem_set_evaluation(a, nf, prec + cond);
     } while(1);
-
-    /* we should not get here */
```
Comment (saraedum) [inline renf_elem/sgn.c:48]: exactly, and the compiler knows that :)

File: e-antic/renf_elem.h (hunk 5 above)
Comment (videlec) [reply e-antic/renf_elem.h:99]: This is one mistake, the other one is that `slong b` should be `ulong b`.

File: poly_extra/sgn_at_half.c (hunk 7 above)
Comment (videlec) [reply poly_extra/sgn_at_half.c:23]: This could be used for an optimization in root finding.

File: renf/randtest.c (hunk 8 above)
Comment (videlec) [reply renf/randtest.c:58]: None.

File: e-antic/e-antic.h.in (hunk 3 above)
Comment (videlec) [reply e-antic/e-antic.h.in:25]: I followed flint here https://github.com/wbhart/flint2/blob/trunk/flint.h#L57

File: poly_extra/evaluate.c (hunk 4 above)
Comment (videlec) [reply poly_extra/evaluate.c:19]: Indeed, the template in `poly_evaluate_template.h` does not need `add` (only `add_fmpz`).

**Pull request merged**